plugins/Oscilloscope/OscilloscopeGraph.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+* OscilloscopeGraph.cpp - Oscilloscope graph widget to handle drawing the waveform and user scrolling/zooming
+*
+* Copyright (c) 2025 Keratin
+*
+* This file is part of LMMS - https://lmms.io
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 2 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program (see COPYING); if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA.
+*
+*/
+
+#include "OscilloscopeGraph.h"
+#include "OscilloscopeControls.h"
+#include "Oscilloscope.h"
+#include "embed.h"
+#include "GuiApplication.h"
+#include "MainWindow.h"
+
+#include <QPainter>
+#include <QSizePolicy>
+#include <QWheelEvent>
+
+#include <functional>
+
+namespace lmms::gui
+{
+
+OscilloscopeGraph::OscilloscopeGraph(QWidget* parent, OscilloscopeControls* controls):
+	QWidget(parent),
+	m_controls(controls),
+	m_inputBufferReader(static_cast<Oscilloscope*>(m_controls->effect())->inputBuffer())
+{
+	setAutoFillBackground(true);
+	setMinimumSize(400, 200);
+	setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+
+	connect(getGUI()->mainWindow(), SIGNAL(periodicUpdate()), this, SLOT(update()));
+	connect(Engine::audioEngine(), &AudioEngine::sampleRateChanged, this, &OscilloscopeGraph::changeSampleRate);
+	changeSampleRate();
+}
+
+void OscilloscopeGraph::changeSampleRate()
+{
+		const auto newBufferSize = static_cast<std::size_t>(Engine::audioEngine()->outputSampleRate() * MaxBufferLengthSeconds);
```

Comment:
```suggestion
	const auto newBufferSize = static_cast<std::size_t>(Engine::audioEngine()->outputSampleRate() * MaxBufferLengthSeconds);
```
removed an extra tab